File: cpp/leetcode/最大子数组和/1.cpp
#include <iostream>
#include <algorithm>
#include <vector>

using namespace std;

class Solution {
public:
    int MaxSubArray(vector<int>& nums) {
        int current_sum = nums[0];
        int max_sum = nums[0];

        for (size_t i = 1; i < nums.size(); ++i) {
            current_sum = max(nums[i], current_sum + nums[i]);
            max_sum = max(max_sum, current_sum);
        }
        return max_sum;
    }
};

int main() {
    Solution solution;
    vector<int> nums = {-2, 1, -3, 4, -1, 2, 1, -5, 4};
    int result = solution.MaxSubArray(nums);
    cout << "Maximum subarray sum: " << result << endl;

    return 0;
}

File: cpp/leetcode/缺失的第一个正数/1.py
from typing import List

class Solution:
    def firstMissiongPositive(self, nums: List[int]) -> int:
        n = len(nums)

        for i in range(n):
            while nums[i] >= 1 and nums[i] <= n and nums[nums[i] - 1] != nums[i]:
                nums[i], nums[nums[i] - 1] = nums[nums[i] - 1], nums[i]
        
        for i in range(n):
            if nums[i] != i + 1:
                return i + 1

        return n + 1


if __name__ == "__main__":
    solution = Solution()
    nums = [1, 2, -1]
    result = solution.firstMissiongPositive(nums)
    print(result)